blog/2022-11-use-define-for-class-fields/README.md

```diff
@@ -0,0 +1,227 @@
+---
+title: 'TypeScript: useDefineForClassFields – How to avoid future Breaking Changes'
+author: Johannes Hoppe and Ferdinand Malcher
+mail: team@angular.schule
+published: 2022-11-29
+lastModified: 2022-11-29
+keywords:
+  - Angular
+  - JavaScript
+  - ECMAScript
+  - TypeScript
+  - ES2022
+  - Class Properties
+  - useDefineForClassFields
+language: en
+thumbnail: usedefineforclassfields.jpg
+sticky: false
+---
+
+
+Did you know that properties are implemented slightly differently in JavaScript and TypeScript and that there is an incompatible behaviour?
+That's why projects with Angular 15 have the option `useDefineForClassFields` set in their TypeScript configuration.
+We will show you in detail what the problem is and how you should write your code so that it is future-proof for both programming languages.
+
+
+## Contents
+
+* [Initialising properties with TypeScript](/blog/2022-11-use-define-for-class-fields#initialising-properties-with-typescript)
+* [The proprietary behaviour of TypeScript](/blog/2022-11-use-define-for-class-fields#the-proprietary-behaviour-of-typescript)
+* [Initialise properties in a future-proof way](/blog/2022-11-use-define-for-class-fields#initialise-properties-in-a-future-proof-way)
+* [Implications for existing Angular code](/blog/2022-11-use-define-for-class-fields#implications-for-existing-angular-code)
+
+
+## Property initialisation in TypeScript
+
+When working with Angular, we regularly initialise properties in our classes.
+For example, a class property can be initialised with a value directly when it is declared.
+There is also a shorthand notation that allows us to declare properties automatically via the constructor. 
+We normally use this short form in Angular to request dependencies through DI (dependency injection).
+
+```ts
+class User {
+  // direct initialisation
+  age = 25;
+
+  // short form
+  constructor(private currentYear: number) {}
+}
+```
+
+## The proprietary behaviour of TypeScript
+
+
+These two notations shown before are proprietary features of TypeScript and have existed since the earliest versions of the language.
+The JavaScript programming language ( or more correctly, the ECMAScript standard) did not fully support such class properties at that time, since the standardisation was still in progress.
```

Comment:
```suggestion
The JavaScript programming language (or more correctly, the ECMAScript standard) did not fully support class properties at that time, since the standardisation was still in progress.
```